cmd/substreams/pack.go

```diff
@@ -20,12 +24,33 @@ var packCmd = &cobra.Command{
 
 func init() {
 	rootCmd.AddCommand(packCmd)
+	packCmd.Flags().StringP("output-dir", "o", "src/pb", cli.FlagDescription(`
+		Optional flag to specify output directory to output generated .spkg file. If the received <package> argument is a local Substreams manifest file
+		(e.g. a local file ending with .yaml), the output folder will be made relative to it
+	`))
 }
 
 func runPack(cmd *cobra.Command, args []string) error {
+	outputDir := maybeGetString(cmd, "output-dir")
+
+	validOutputDirectorySpecified := true
+	fileInfo, err := os.Stat(outputDir)
+	if err != nil || !fileInfo.IsDir() {
+		fmt.Println("WARNING: Output directory specified is invalid - falling back to default path!\nOutputDir=", outputDir)
+		fmt.Println("")
+		validOutputDirectorySpecified = false
+	}
+
 	manifestPath := args[0]
 
 	manifestReader := manifest.NewReader(manifestPath)
+
+	if validOutputDirectorySpecified && manifestReader.IsLocalManifest() && !filepath.IsAbs(outputDir) {
```

Comment:
Definitely want to be able to specify relative directory paths as it's a pain to have to work with absolute paths cross-platform when calling it from places like makefiles. Was new to golang do didn't realise what the right command was but essentially we'd need to check that for a folder specified, is it specified from the root folder of your operating system or not. If not then we will treat it like a relative path. I guess I wouldn't really want to decide whether it's a relative path or not depending on whether the manifest was local anyway.

Reply:
I guess a simple way to do this would be just to check whether the first character of the supplied path is '.' or not.. what would you think about doing that?

Reply:
`filepath.IsAbs(<path>)` can be used to determine if a path is relative already or absolute. Relative path handling will of course continue to work, the question is more abou what is the best default behavior for `substreams pack`:

- Relative to `manifest.yaml` file
- Relative to working directory of the process (so usually the path in your terminal where the command is invoked).

I think for `substreams pack`, the best default is relative to command execution, scripting wise it's the easiest. It's also the easiest to reason about when passing the flag, specially for project with multiple substreams:

```
substreams pack -o build/spkgs substreams1/manifest.yaml
```

Is it `./build/spkgs` or `substreams1/build/spkgs/manifest.yaml`? I had this debate internally even for `substreams protogen` but in this particular case, it made sense to make it relative to `manifest.yaml` (when `IsLocalmanifest`) because the generated protobuf files are meant to be used in this particular substreams project.

Not so much the case with `pack`.

Reply:
I've made what I think is a good change if you want to have a look? Think this should be the clearest way to deal with this.